CMakeLists.txt

```diff
@@ -29,36 +28,136 @@ set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY
 list( APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" )
 include( traccc-compiler-options )
 
-# option for algebra plugins (ARRAY EIGEN SMATRIX VC VECMEM)
-set(TRACCC_ALGEBRA_PLUGIN EIGEN)
-if (TRACCC_ALGEBRA_PLUGIN)
-  message(STATUS "Building with plugin type: " ${TRACCC_ALGEBRA_PLUGIN})
-  set(ALGEBRA_PLUGIN_INCLUDE_${TRACCC_ALGEBRA_PLUGIN} ON)
-  add_definitions(-DALGEBRA_PLUGIN_INCLUDE_${TRACCC_ALGEBRA_PLUGIN})
+# Check if CUDA is available.
+include( CheckLanguage )
+check_language( CUDA )
+set( TRACCC_BUILD_CUDA_DEFAULT FALSE )
+if( CMAKE_CUDA_COMPILER )
+   set( TRACCC_BUILD_CUDA_DEFAULT TRUE )
 endif()
 
-# option for algebra scalar type 
-set(TRACCC_CUSTOM_SCALARTYPE float)
-if(TRACCC_CUSTOM_SCALARTYPE)
-    message(STATUS "Building with custom scalar type: " ${TRACCC_CUSTOM_SCALARTYPE})
-    set(ALGEBRA_PLUGIN_CUSTOM_SCALARTYPE ${TRACCC_CUSTOM_SCALARTYPE})
```

Comment:
Is there a reason we're getting rid of this functionality?

Reply:
It is all set through `ALGEBRA_PLUGINS_CUSTOM_SCALARTYPE` now. Which is a cache variable in algebra-plugins. Since it's a cache variable, we don't have to define it here. We can override it with `-DALGEBRA_PLUGINS_CUSTOM_SCALARTYPE=float` from the command line, if we absolutely need to.

But as I've been advertising to @niermann999 and @beomki-yeo for the last weeks, I believe making this choice during the CMake configuration is a bad choice to begin with. I will want to re-design the algebra-plugins and detray projects to allow the user deciding about the scalar type as a template parameter. So that they could try out "detector objects" with different scalar types side-by-side, in the same source file if they wanted to.

So removing the setting from here is a precursor for things to come...

Reply:
Fully agree on making this a template parameter. :+1: